src/DayColumn.js

```diff
@@ -271,15 +271,23 @@ let DaySlot = React.createClass({
     selector
       .on('click', (box) => {
         if (!isEvent(findDOMNode(this), box))
-          this._selectSlot(selectionState(box))
+          this._selectSlot(Object.assign({},
+            selectionState(box), {
+              action: 'click'
+            }
+          ))
 
         this.setState({ selecting: false })
       })
 
     selector
       .on('select', () => {
         if (this.state.selecting) {
-          this._selectSlot(this.state)
+          this._selectSlot(Object.assign({},
+            this.state, {
+              action: 'select'
+            }
```

Comment:
`{ ...this.state, action: 'select' }` is a bit neater instead of the Object.assign